server/routes/api/users.ts

```diff
@@ -184,6 +186,16 @@ router.post("users.update", auth(), async (ctx) => {
   if (language) {
     user.language = language;
   }
+  if (preferences) {
+    assertKeysIn(preferences, UserPreference);
+    if (preferences.lastVisitedPath) {
+      assertUrlPath(preferences.lastVisitedPath);
```

Comment:
Lets just make this a boolean, we don't want to have to write the user record every time the path changes – it'll just be stored on the client.